### ISSUE TITLE
After update to latest version 1.14.2 errors

### ISSUE THREAD

**Issue by PvdGulik** [OpenWonderLabs/homebridge-switchbot#444]
### Describe The Bug

repeating TypeError message: processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')

Stopping de Hombridge Switcht plugin stops the errors in the logging.

### To Reproduce

Update from 1.14.1  ->> 1.14.2

### Expected behavior

no TypeError

### Relevant log output

```shell
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')```


### Config for homebridge-switchbot

```shell
{
    "name": "SwitchBot",
    "credentials": {
        "openToken": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
        "notice": "Keep your token a secret!"
    },
    "options": {
        "devices": [
            {
                "deviceId": "E3:25:5E:A7:XX:XX",
                "configDeviceName": "Curtain 90",
                "configDeviceType": "Curtain",
                "ble": true,
                "curtain": {
                    "set_minStep": 20,
                    "setOpenMode": "0",
                    "setCloseMode": "0"
                },
                "logging": "standard"
            }
        ],
        "logging": "standard"
    },
    "platform": "SwitchBot"
}
```

### Device & Model

Curtain 90

### Node.js Version

v16.17.0

### NPM Version

8.15.0

### Homebridge Version

v1.5.0

### Homebridge Switchbot Plugin Version

v1.14.2

### Homebridge Config UI X Plugin Version

v4.50.1

### Operating System

Raspian

**Comment by grunkyb** [comment]
I have this error too and it persists when returning to 1.14.0

**Comment by donavanbecker** [comment]
None of those errors look like they are coming from SwitchBot

**Comment by PvdGulik** [comment]
Hello Donavan,

That is a little fast concluded. After upgrading to the latest plugin it started with this error.
And when you dis-enable the Switcht plugin, the error disappears. so how can you tell "None of those errors look like they are coming from SwitchBot.

Thanks!

greet Pieter

**Comment by zaki-hanafiah** [comment]
Can confirm as well, got the same after updating to 1.14.2, but it seems to persist even if rolling back to the older version.
Disabling the plugin stops the log from spamming.

![image](https://user-images.githubusercontent.com/31648832/185790829-6fb10de3-dc53-423c-a2bd-a1364b21044c.png)

**Comment by PvdGulik** [comment]
Hello Donavan,

Please reopen this issue, there are more people having this issue.

Thanks

vriendelijke groet,

Pieter

**Comment by grunkyb** [comment]
SwitchBot are the only BLE devices I’m running from Homebridge

**Comment by zaki-hanafiah** [comment]
Same with @grunkyb  here.

**Comment by PvdGulik** [comment]
Same as @zaki and @grunkyb Switchbot is my only ble device in Homebridge....

**Comment by donavanbecker** [comment]
@PvdGulik I am not experiencing this and that is why it is closed as cannot reproduce.

**Comment by TommyGKiwi** [comment]
> @PvdGulik I am not experiencing this and that is why it is closed as cannot reproduce.

I am also experiencing this as well after update, seems it might be an API you use as even rolling back, or switching to the other Verified Switchbot plug-in that is Bluetooth only has the same error. Only disabling the plug-in the issue goes away.

**Comment by teh-hippo** [comment]
I'm having the same issue, and v1.14.2 seems to be the cause.

**Comment by galaxydpw** [comment]
same issue here.
rollback to 1.14.1 makes the error disappear for me.

**Comment by grunkyb** [comment]
Here's how it looks with my debug logs turned on for SwitchBot

```
[22/08/2022, 17:55:35] [SwitchBot] 01:23:45:67:89:ab
[22/08/2022, 17:55:35] [SwitchBot] 01:23:45:67:89:ab
[22/08/2022, 17:55:35] [SwitchBot] Humidifier: Humidifier BLE Address Found: 01:23:45:67:89:ab
[22/08/2022, 17:55:35] [SwitchBot] Humidifier: Humidifier Config BLE Address: 01:23:45:67:89:ab
[22/08/2022, 17:55:35] [SwitchBot] [DEBUG] Humidifier: Humidifier serviceData: {"model":"e","modelName":"WoHumi","onState":true,"autoMode":false,"percentage":27}
[22/08/2022, 17:55:35] [SwitchBot] [DEBUG] Humidifier: Humidifier model: e, modelName: WoHumi,autoMode: false, onState: true, percentage: 27
[22/08/2022, 17:55:35] [SwitchBot] [DEBUG] Humidifier: Humidifier connected: true
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
```

<details>
<summary>Show more of the log</summary>

```
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
```
</details>

This repeats itself a few times with varying numbers of repeats of the TypeError then this
```
[22/08/2022, 17:55:43] [SwitchBot] 01:23:45:67:89:ab
[22/08/2022, 17:55:43] [SwitchBot] 01:23:45:67:89:ab
[22/08/2022, 17:55:43] [SwitchBot] Humidifier: Humidifier BLE Address Found: 01:23:45:67:89:ab
[22/08/2022, 17:55:43] [SwitchBot] Humidifier: Humidifier Config BLE Address: 01:23:45:67:89:ab
[22/08/2022, 17:55:43] [SwitchBot] [DEBUG] Humidifier: Humidifier serviceData: {"model":"e","modelName":"WoHumi","onState":true,"autoMode":false,"percentage":27}
[22/08/2022, 17:55:43] [SwitchBot] [DEBUG] Humidifier: Humidifier model: e, modelName: WoHumi,autoMode: false, onState: true, percentage: 27
[22/08/2022, 17:55:43] [SwitchBot] [DEBUG] Humidifier: Humidifier connected: true
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
```
<details>
<summary>
Show more of the log
</summary>

```
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
processLeAdvertisingReport: Caught illegal packet (buffer overflow): TypeError: Cannot read properties of undefined (reading 'slice')
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier BLE parseStatus
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier CurrentRelativeHumidity: 27
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier Active: 1
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier updateCharacteristic CurrentRelativeHumidity: 27
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier updateCharacteristic CurrentHumidifierDehumidifierState: 0
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier updateCharacteristic TargetHumidifierDehumidifierState: 1
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier updateCharacteristic Active: 1
[22/08/2022, 17:55:44] [SwitchBot] [DEBUG] Humidifier: Humidifier updateCharacteristic RelativeHumidityHumidifierThreshold: 0
```
</details>

**Comment by PvdGulik** [comment]
Hello Donavan, Please reopen this issue, as you can see there are more and more people having this issue. Thanks

**Comment by grunkyb** [comment]
I'm able to connect to the device through the OpenAPI, e.g.,` curl -X GET -k -H 'Authorization: xxxx' -i 'https://api.switch-bot.com/v1.0/devices/0123456789AB/status'`

What's everyone using here? I'm on a Raspberry Pi 4B running 64-bit Bullseye.

I'm wondering if it's a Pi + BLE thing but need others to confirm.

**Comment by TommyGKiwi** [comment]
For me at least, you can still activate the Switchbot, but there is a lot of spam in the log. 
I have pi 3B and using BLE
I also noticed there were some build errors In the post upgrade process with a 404 error for abandonware, but that might be a red herring

**Comment by zaki-hanafiah** [comment]
Like @TommyGKiwi  mentioned, I can also still use my Switchbot and the existing automations all work fine, it's just the issue with the errors that are spamming the homebridge logs.

**Comment by PvdGulik** [comment]
Thanks for reopening.

For me the SwitchBot works, just the spam in the log. I’m using a Raspberry Pi 3B+

Thanks

Pieter

**Comment by CyberMrProper** [comment]
I'm having the same issue in version 1.14.2

**Comment by galaxydpw** [comment]
> same issue here.
> rollback to 1.14.1 makes the error disappear for me.

I reported the error in my lab env, a pi zero 2, with raspbian homebridge and nothing else.

However, I did the installation of the plugin in my prod env, pi4, with many other plugins, and I don't have the error.

**Comment by donavanbecker** [comment]
Seems to only be a raspberryPi [noisily](https://github.com/OpenWonderLabs/node-switchbot/pull/142#issuecomment-1229187369) log issue.

**Comment by PvdGulik** [comment]
I found that during the installation I get the following error:

node-pre-gyp ERR! install response status 404 Not Found on https://github.com/abandonware/node-bluetooth-hci-socket/releases/download/0.5.3-8/bluetooth_hci_socket-0.5.3-8-node-v93-linux-arm.tar.gz 


When I browse following that link, there's nothing in [https://github.com/abandonware/node-bluetooth-hci-socket/releases/](https://github.com/abandonware/node-bluetooth-hci-socket/releases/).
Is this related to the Issue #444?

Thanks

**Comment by donavanbecker** [comment]
This is a node-SwitchBot/noble issue. It will have to resolve at those repos before it can be resolved here.

**Comment by donavanbecker** [comment]
Has anyone tried this: https://github.com/OpenWonderLabs/homebridge-switchbot/issues/425#issuecomment-1190864279

**Comment by grunkyb** [comment]
Yes, I had to use that to connect Pi to BLE.

**Comment by grunkyb** [comment]
Can anyone spot anything in [this function](https://github.com/abandonware/noble/blob/master/lib/hci-socket/hci.js#L1109) in noble? You can see that this could be throwing it.

```
Hci.prototype.processLeAdvertisingReport = function (numReports, data) {
  try {
    for (let i = 0; i < numReports; i++) {
      const type = data.readUInt8(0);
      const addressType = data.readUInt8(1) === 0x01 ? 'random' : 'public';
      const address = data
        .slice(2, 8)
        .toString('hex')
        .match(/.{1,2}/g)
        .reverse()
        .join(':');
      const eirLength = data.readUInt8(8);
      const eir = data.slice(9, eirLength + 9);
      const rssi = data.readInt8(eirLength + 9);

      debug(`\t\t\ttype = ${type}`);
      debug(`\t\t\taddress = ${address}`);
      debug(`\t\t\taddress type = ${addressType}`);
      debug(`\t\t\teir = ${eir.toString('hex')}`);
      debug(`\t\t\trssi = ${rssi}`);

      this.emit(
        'leAdvertisingReport',
        0,
        type,
        address,
        addressType,
        eir,
        rssi
      );

      data = data.slice(eirLength + 10);
    }
  } catch (e) {
    console.warn(
      `processLeAdvertisingReport: Caught illegal packet (buffer overflow): ${e}`
    );
  }
};
```

**Comment by csyee** [comment]
I have the same error too.

**Comment by PvdGulik** [comment]
> Has anyone tried this: https://github.com/OpenWonderLabs/homebridge-switchbot/issues/425#issuecomment-1190864279

Yes, but did not resolve the issue..
Still continues errors in the logging

**Comment by Niklas31** [comment]
I have the same issue. It switches from local BLE to mini hub, therefore, no operation problem, but log are full of this error.

**Comment by nyjklein** [comment]
Not sure why there seems to be two bug reports for the same issue, but I'll chime in. I have the same issue (Raspberry Pi). I suspected SwitchBot since it started after I upgraded to 1.14.2 but when I rolled back the issue persisted. So I wasn't sure. Then I was away on vacation for two weeks. Came back to the issue still happening and I see others have reported the issue here.

**Comment by grunkyb** [comment]
I've done an error stack trace on this and found a workaround.

I changed lines 78–80 in `OpenWonderLabs/node-switchbot/blob/latest/lib/switchbot-advertising.js` from
```
    if (bufIsInvalid && manufacturerDataIsInvalid) {
      return null;
    }
```
to
```
    if (bufIsInvalid || manufacturerDataIsInvalid) {
      return null;
    }
```

If you look at the blame for the code, you can see that the `&& manufacturerDataIsInvalid` was added on 18 August, right about when this problem started showing up.

On my RPi, the path to the file to change is `/var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/node_modules/node-switchbot/lib`.

Can anyone else check this is the right solution?

**Comment by donavanbecker** [comment]
@grunkyb, I will make this change and then push it out to a beta. can you test it then? it will be part of Issue #476 & Pull Request #474

**Comment by donavanbecker** [comment]
@grunkyb latest be there has these changes.

**Comment by grunkyb** [comment]
Installed v1.15.1-beta.11  early this morning and no more error logs. API 1.1 working well too.

**Comment by zaki-hanafiah** [comment]
Same here, `1.15.1-beta.11` has resolved the noisy logs for me. Everything works accordingly as well. No regressions to report.

**Comment by donavanbecker** [comment]
Perfect! I will get 2.0.0 all packaged up and publish it soon.

**Comment by PvdGulik** [comment]
I don't know how to test, but I see it is solved in 2.0.0. When will 2.0.0 be released?

**Comment by donavanbecker** [comment]
still have a few changes to make but beta is available: #476

**Comment by PvdGulik** [comment]
Oke, I installed 2.0.0 but found I only can select Bot, no Curtain. Is that supposed to be so? 

Thanks!

**Comment by donavanbecker** [comment]
> Oke, I installed 2.0.0 but found I only can select Bot, no Curtain. Is that supposed to be so? 
> 
> Thanks!

Do you have a screenshot?

**Comment by PvdGulik** [comment]
![image](https://user-images.githubusercontent.com/9421474/194415961-ad45483a-7078-4adc-b0a5-9927190bfc00.jpeg)

**Comment by donavanbecker** [comment]
that isn't this plugin.

**Comment by PvdGulik** [comment]
Sorry, you're right. I installed the wrong one. Now the right-one, version 2.0.0-beta.19

And I can confirm problem solved!👌

**Comment by donavanbecker** [comment]
Fixed with v2.0.0, Once released.
